x/accesscontrol/keeper/keeper.go

```diff
@@ -250,6 +250,26 @@ func BuildSelectorOps(accessOps []acltypes.AccessOperationWithSelector, senderBe
 				opWithSelector.Operation.IdentifierTemplate,
 				hexStr,
 			)
+		case acltypes.AccessOperationSelectorType_CONTRACT_REFERENCE:
+			// We use this to import the dependencies from another contract address
+			interContractAddress, err := sdk.AccAddressFromBech32(opWithSelector.Selector)
+			if err != nil {
+				return []acltypes.AccessOperationWithSelector{}, err
+			}
+			// TODO: add a circular dependency check here to ignore if we've already seen this contract/identifier in our reference chain
+			// for now, we will just pass in the same message body, this needs to be changed later though
+			// TODO: build new msgbody for the new contract execute / query msg in later milestone tasks
+			wasmDeps, err := k.GetWasmDependencyMapping(ctx, interContractAddress, contractAddr.String(), msgBody, true)
+
+			if err != nil || !wasmDeps.Enabled {
+				// if we have an error fetching the dependency mapping or the mapping is disabled, we want to use the synchronous mappings instead
+				selectedAccessOps = append(selectedAccessOps, types.SynchronousAccessOpsWithSelector()...)
+			} else {
+				// if we did get deps properly and they are enabled, now we want to add them to our access operations
+				selectedAccessOps = append(selectedAccessOps, wasmDeps.AccessOps...)
```

Comment:
will this significantly increase the size of accessOps list, suppose there are a contract depends on 100 other contracts then it will have an accessOps list 100x bigger than the others. Maybe we can use a Map instead of List in the future?

Reply:
this is true, we can apply a step of deduplicating identical access operations in the future, but I don't think its a significant issue right now. Good call out though, I'lll create a task for the future to do this

Reply:
sg can we add a TODO here

Reply:
done